### PR TITLE
Close #333: Introduce brand new URI matching in Router

### LINKED ISSUE #333: Current Path have no possibilities to work with path-parameters
For some sort of web-application is very important to have possibility share "direct link to anywhere"to achieve that only Path parameters is not enough

### PR DESCRIPTION
**Pull request by greenhost87**

### CODE REVIEW

**Comment by fomkin** [comment]
More elegant approach https://scastie.scala-lang.org/vuZsWmJYRpedVvSBSKMdVw

File: modules/web/src/main/scala/korolev/web/PathAndQuery.scala [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    @tailrec
+    def unapply(pq: PathAndQuery): Option[(Path, String)] = pq match {
+      case slash: /   => Some(slash.prev -> slash.value)
+      case path :? _  => unapply(path)
+      case query :& _ => unapply(query)
+      case _          => None
+    }
+  }
+
+  final case class =?=(k: String, v: String) extends Query
```
Comment (fomkin) [inline modules/web/src/main/scala/korolev/web/PathAndQuery.scala:166]: Replace with Tuple.

File: modules/web/src/main/scala/korolev/web/PathAndQuery.scala [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      }
+    }
+
+    aux(this, tail.reverse)
+  }
+}
+
+object PathAndQuery {
+  case object Root extends Path
+  case object RelativeRoot extends Path
```
Comment (fomkin) [inline modules/web/src/main/scala/korolev/web/PathAndQuery.scala:149]: We discussed this before. RelativeRoot exists only in browsers `src=""` URI resolvers. I do not understand why we should add this hierarchy. Could you show use cases?

Comment (greenhost87) [reply modules/web/src/main/scala/korolev/web/PathAndQuery.scala:149]: Drop RelativeRoot

File: modules/korolev/src/main/scala/korolev/package.scala [hunk 3]
```diff
@@ -19,8 +19,12 @@ import korolev.web.Path
...
 
   // Routing API
-  @inline val Root = Path.Root
-  @inline val / = Path./
+  @inline val Root: PathAndQuery.Root.type = PathAndQuery.Root
+  @inline val / : PathAndQuery./.type = PathAndQuery./
+  @inline val :? : PathAndQuery.:?.type = PathAndQuery.:?
+  @inline val :?? : PathAndQuery.:??.type = PathAndQuery.:??
+  @inline val :?* : PathAndQuery.:?*.type = PathAndQuery.:?*
+  @inline val *& : PathAndQuery.*&.type = PathAndQuery.*&
```
Comment (fomkin) [inline modules/korolev/src/main/scala/korolev/package.scala:28]: Add `:&`

**Pull request merged**